ui/guest/src/classes/ContentController.ts

```diff
@@ -241,12 +241,11 @@ export function insertItem(modelId: string, fieldId: string, index: number | str
   // Insert in desired position
   result.splice(index, 0, item);
 
+  setCollection(model, fieldId, typeof index === 'string' ? removeLastPiece(index) : index, result);
```

Comment:
- Index may not have `.`, even when a string. `removeLastPiece` will return an empty string in that scenario.
- It's important we don't mutate existing objects. The setCollection chain looks suspicious, like it would mutate the containing collections of the target one. We need to spread/create-new objects for each of the pieces of the nested prop/collection.
- Perhaps the method(s) rather belong to utils/object